### PR TITLE
Feature - eCommerce

### PR DESCRIPTION
**Pull request by aulisius**
## Proposed changes

This adds the eCommerce module to the plugin. Users of the plugin who are the on the eCommerce plans will be redirected to the new dashboard instead of the current `/home/onboarding` screens.

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Pending Items

- [ ] Brand specific CSS tokens can be loaded via the module itself.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "author": "David Ryan <dryanpress@chat.wordpress.org>",
     "license": "GPL-2.0-or-later",
     "dependencies": {
+        "@newfold-labs/wp-module-ecommerce": "newfold-labs/wp-module-ecommerce#v0.2.0",
```
Comment (wpscholar) [inline package.json:16]: What exactly is this dependency referencing? I see the module package.json points to `dist/index.js` (https://github.com/newfold-labs/wp-module-ecommerce/blob/trunk/package.json#L7), but that doesn't exist in the repo and I don't see any packages released (https://github.com/newfold-labs/wp-module-ecommerce/packages).

Comment (aulisius) [reply package.json:16]: npm lets you specify git sources as package url. You can refer to the [npm install](https://docs.npmjs.com/cli/v7/commands/npm-install#:~:text=npm%20install%20%3Cgithubname%3E/%3Cgithubrepo%3E%5B%23%3Ccommit%2Dish%3E%5D%3A) documentation for more details.

npm does the following things for the given package

```json
"@newfold-labs/wp-module-ecommerce": "newfold-labs/wp-module-ecommerce#v0.2.0"
```

- Search for newfold-labs/wp-module-ecommerce repo on GitHub
- Check if repo has a commit ref or tag with the value `v0.2.0`
- Download code from that commit
- Check if downloaded `package.json` file has a `prepare` script.
- Trigger the `prepare` script
- Clean up remaining files based on what `files` targets in `package.json`

In our case, the `prepare` script triggers the commands to produce the `dist/` folder which contains the needed build artifacts.

File: src/app/store/selectors.js [hunk 2]
```diff
@@ -25,6 +25,16 @@ export function isTopLevel(state) {
...
 }
 
+export function isOnECommercePlan(state) {
+  return ["wc_standard", "wc_premium"].includes(
+    state.app.customer.plan_subtype
+  );
+}
+
+export function isNewEcommerceUser(state) {
+	return new Date(state.app.customer.signup_date) >= new Date('2022-08-18T17:00:00.000Z');
```
Comment (0aveRyan) [inline src/app/store/selectors.js:35]: @aulisius Per a meeting with Jocelyn just now, we've picked a date/time of August 18 at 10AM Mountain Time I pushed here now. This is something we'll be addressing in Onboarding as well.
Screenshot 2022-08-03 @ 03 02 29 PM

Comment (aulisius) [reply src/app/store/selectors.js:35]: Understood.

**Review by wpscholar — CHANGES_REQUESTED** [review]
@aulisius I've created a new [feature/e-commerce](https://github.com/bluehost/bluehost-wordpress-plugin/tree/feature/ecommerce) branch in the main Bluehost repo. I've already merged the latest version of master into that branch. Since we've recently updated to a new version of React Router DOM, there may be an issue with the e-commerce onboarding page not loading (I couldn't see any of the new functionality except the "Site Status" in the admin bar).

So, here are a few things to do:

- Convert this PR to a draft
- Pull down the latest changes from the Bluehost `feature/ecommerce` branch to your local `feature/ecommerce` branch.
- Update your PR to merge into the Bluehost `feature/ecommerce` branch instead of `master`.
- Look into why the e-commerce onboarding page isn't displaying
- Update the README.md to include what steps are required to trigger the e-commerce onboarding flow from a development perspective. For example, making sure the install date is a particular value, the database information for plan type is set, etc.
- Update the DTS build to use wp-scripts
- Address any remaining TODOs in the code (I'm seeing [this](https://github.com/newfold-labs/wp-module-ecommerce/blob/997b0570bf78692a4edb0991f15def0c043207d4/src/components/CustomizeStore.js#L33))
- Tag a new release of the e-commerce module once you're done making changes
- Update the e-commerce module version in the Bluehost plugin
- Move this PR from draft to ready for review

**Comment by aulisius** [comment]
@wpscholar regarding `Update the DTS build to use wp-scripts`, Does `wp-scripts` let you emit libraries instead of generated bundles that require to be loaded separately by PHP?

**Comment by wpscholar** [comment]
@aulisius Correct.

Here is the documentation for wp-scripts: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/

**Pull request merged**